### PR TITLE
Update IR for equations

### PR DESCRIPTION
**Pull request by HarrisonGrodin**

### CODE REVIEW

File: src/function_registration.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ for (M, f, arity) in DiffRules.diffrules()
     @eval @register $sig
 end
 
-for fun = (:<, :>, :(==), :~, :!, :&, :|, :div)
```
Comment (ChrisRackauckas) [inline src/function_registration.jl]: ahh yes, I could see that causing issues.

File: src/systems/diffeqs/diffeqsystem.jl [hunk 2]
```diff
@@ -41,71 +41,64 @@ function generate_ode_function(sys::DiffEqSystem;version = ArrayFunction)
...
+
+    return :($lhs = $(convert(Expr, eq.rhs)))
 end
 
-function calculate_jacobian(sys::DiffEqSystem,simplify=true)
-    diff_idxs = map(eq->eq.args[1].diff !=nothing,sys.eqs)
-    diff_exprs = sys.eqs[diff_idxs]
-    rhs = [eq.args[2] for eq in diff_exprs]
+function calculate_jacobian(sys::DiffEqSystem, simplify=true)
+    calcs, diff_exprs = partition(isintermediate, sys.eqs)
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/diffeqsystem.jl]: what's this `partition` function?

File: src/utils.jl [hunk 3]
```diff
@@ -33,6 +33,12 @@ end
 
 toexpr(ex) = MacroTools.postwalk(x -> isa(x, Expression) ? convert(Expr, x) : x, ex)
 
+function partition(f, xs)
```
Comment (ChrisRackauckas) [inline src/utils.jl]: found it.

File: src/systems/diffeqs/diffeqsystem.jl [hunk 4]
```diff
@@ -144,5 +137,44 @@ function DiffEqBase.ODEFunction(sys::DiffEqSystem;version = ArrayFunction,kwargs
     end
 end
 
+function extract_elements(eqs, targetmap, default = nothing)
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/diffeqsystem.jl]: why was this moved here?

Comment (HarrisonGrodin) [reply src/systems/diffeqs/diffeqsystem.jl]: At one point, it wouldn't precompile because I was dispatching on `Equation` there before it was defined. This shouldn't be an issue anymore, though, since I switched to `isa`. Would you like me to move it back?

Comment (HarrisonGrodin) [reply src/systems/diffeqs/diffeqsystem.jl]: Ah, good point. Done.
